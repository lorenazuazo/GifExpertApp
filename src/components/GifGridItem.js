import React from 'react';
import PropTypes from 'prop-types';
import 'animate.css/animate.min.css';

const GifGridItem = ({id,title,url}) => {

    return (
        <div className="card animate__animated animate__fadeInRight">
            <img src={url} alt={title}></img>
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}

export default GifGridItem;
