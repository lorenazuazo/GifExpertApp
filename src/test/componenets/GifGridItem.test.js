import React from 'react';
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Test Suite component <GifGridItem />', () => {
    
    const img = {
        id :'1',
        title:'Test al componente',
        url:'https://localhost/algo.jpg'
    }

    const wrapper = shallow(<GifGridItem {...img} />);
    test('El componente se debe mostrar correctamente}', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe tener un parrafo con el valor del props title ', () => {

        const p = wrapper.find('p').text().trim();
        expect(p).toBe(img.title);
    });
    test('debe tener la imagen igual a la url y el alt pasados por props', () => {
        const imagen=wrapper.find('img');
        //console.log(imagen.prop('src'))
        expect(imagen.prop('src')).toBe(img.url);
        expect(imagen.prop('alt')).toBe(img.title)
    });

    test('que el div contenga className=animate__fadeInRight', () => {
        const clase='animate__fadeInRight';
        const claseWrapper=wrapper.find('div').prop('className')
        expect(claseWrapper).toContain(clase)
        //tambien se puede hacer
        //expect(claseWrapper.includes(clase)).not.toBe(false)
        //expect(claseWrapper.includes(clase)).toBe(true)
        
    });
    
})
