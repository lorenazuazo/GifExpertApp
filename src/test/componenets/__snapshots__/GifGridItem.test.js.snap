// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Suite component <GifGridItem /> El componente se debe mostrar correctamente} 1`] = `
<div
  className="card animate__animated animate__fadeInRight"
>
  <img
    alt="Test al componente"
    src="https://localhost/algo.jpg"
  />
  <p>
    Test al componente
  </p>
</div>
`;
